core/src/test/java/io/aiven/klaw/TopicRequestValidatorImplIT.java

```diff
@@ -0,0 +1,276 @@
+
+    Set<ConstraintViolation<TopicRequestModel>> violations = validator.validate(addTopicRequest);
+    assertThat(violations.size()).isEqualTo(1);
+    assertThat(violations.toString())
+        .contains("Topic prefix does not match. " + addTopicRequest.getTopicname());
+
+    env.setOtherParams("topic.suffix=" + topicPrefixSuffix);
+    when(topicControllerService.getEnvDetails(anyString())).thenReturn(env);
+
+    violations = validator.validate(addTopicRequest);
+    assertThat(violations.toString())
+        .contains("Topic suffix does not match. " + addTopicRequest.getTopicname());
+  }
+
+  @Test
+  @Order(8)
+  public void isValidTestVerifyIfTopicRequestAlreadyExists() {
+    int tenantId = 101;
+    Topic topic = utilMethods.getTopic("testtopic");
+    topic.setTeamId(1001);
+
+    TopicRequestModel addTopicRequest = utilMethods.getTopicRequestModel(1001);
+    TopicRequest topicRequest = utilMethods.getTopicRequest(1001);
+    when(commonUtilsService.isNotAuthorizedUser(any(), any())).thenReturn(false);
+    when(topicControllerService.getUserName()).thenReturn("superadmin");
+    when(topicControllerService.getEnvsFromUserId(any())).thenReturn(List.of("1"));
+    when(commonUtilsService.getTenantId(any())).thenReturn(tenantId);
+    when(topicControllerService.getEnvDetails(anyString()))
+        .thenReturn(utilMethods.getEnvLists().get(0));
+    when(topicControllerService.getExistingTopicRequests(addTopicRequest, tenantId))
+        .thenReturn(List.of(topicRequest));
+
+    Set<ConstraintViolation<TopicRequestModel>> violations = validator.validate(addTopicRequest);
+    assertThat(violations.size()).isEqualTo(1);
+    assertThat(violations.toString()).contains("Failure. A topic request already exists.");
+  }
+
+  @Test
+  @Order(9)
+  public void isValidTestVerifyIfTopicAlreadyExists() {
+    int tenantId = 101;
+    Topic topic = utilMethods.getTopic("testtopic1001");
+    int teamId = 1001;
+    topic.setTeamId(teamId);
+    topic.setEnvironment("1");
+
+    TopicRequestModel addTopicRequest = utilMethods.getTopicRequestModel(teamId);
+    when(commonUtilsService.isNotAuthorizedUser(any(), any())).thenReturn(false);
+    when(topicControllerService.getUserName()).thenReturn("superadmin");
+    when(topicControllerService.getTeamId(anyString())).thenReturn(teamId);
+    when(topicControllerService.getEnvsFromUserId(any())).thenReturn(List.of("1"));
+    when(commonUtilsService.getTenantId(any())).thenReturn(tenantId);
+    when(topicControllerService.getEnvDetails(anyString()))
+        .thenReturn(utilMethods.getEnvLists().get(0));
+    when(topicControllerService.getTopicFromName(anyString(), anyInt())).thenReturn(List.of(topic));
+    when(topicControllerService.getExistingTopicRequests(any(), anyInt()))
+        .thenReturn(Collections.emptyList());
+
+    Set<ConstraintViolation<TopicRequestModel>> violations = validator.validate(addTopicRequest);
+    assertThat(violations.size()).isEqualTo(1);
```

Comment:
```suggestion
    assertThat(violations).hasSize(1);
```
assertJ has `hasSize` it's better to use it here

Reply:
Agree, updated.